docs/examples/zoo/images/decode_and_transform_pytorch.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+import json
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+import nvidia.dali.plugin.pytorch.experimental.proxy as dali_proxy
+from nvidia.dali import pipeline_def, fn, types
+import glob
+from pathlib import Path
+
+
+class ImageDataset(Dataset):
+    def __init__(self, image_dir, json_path, transform=None):
+        """
+        Initialize the dataset.
+
+        :param image_dir: Directory containing the images.
+        :param json_dir: Directory containing the JSON labels.
+        :param transform: Optional transform to apply on images.
+        """
+        self.image_dir = image_dir
+        self.transform = transform
+        self.image_ids = glob.glob("*.jpg", root_dir=image_dir)
+
+        # Load the JSON label
+        with open(json_path, "r") as f:
+            self.labels = json.load(f)
```

Comment:
Do you think it's worth processing this to a dict that maps image_id to label? The lookups would be using the underlying hash map rather than iterating over the whole dataset for every file.